bundler/lib/bundler/feature_flag.rb

```diff
@@ -41,7 +41,7 @@ def self.settings_method(name, key, &default)
     settings_flag(:plugins) { @bundler_version >= Gem::Version.new("1.14") }
     settings_flag(:print_only_version_number) { bundler_3_mode? }
     settings_flag(:setup_makes_kernel_gem_public) { !bundler_3_mode? }
-    settings_flag(:skip_default_git_sources) { bundler_3_mode? }
+    settings_flag(:skip_default_git_sources) { bundler_4_mode? }
```

Comment:
In this version I delayed the deprecation, but I believe there's no use of maintaining code that never gets run by either end users or tests, so I would rather actually remove this code until we release bundler 3, and only then restore it.